File: frontend/src/features/devices/api/updateDevice.ts
import { http } from "../../../shared/api/http";
import { mapDevice } from "./deviceMapper";
import type { UpdateDeviceInput, DeviceRead } from "../model/types";
import type { DeviceApiDto } from "./types";

export async function updateDevice(
  deviceId: string,
  data: UpdateDeviceInput,
): Promise<DeviceRead> {
  const response = await http.put<DeviceApiDto>(`/devices/${deviceId}`, {
    name: data.name,
    serial_number: data.serialNumber,
    registry_number: data.registryNumber,
    modification: data.modification,
    manufacture_year: data.manufactureYear,
    inventory_number: data.inventoryNumber,
    comment: data.comment,
  });

  return mapDevice(response.data);
}

File: backend/app/application/device/commands/update_device.py
"""
Update device command.
"""

from dataclasses import dataclass
from uuid import UUID


@dataclass(frozen=True)
class UpdateDeviceCommand:
    """Command for updating a device card."""

    device_id: UUID
    name: str
    serial_number: str
    registry_number: str | None = None
    modification: str | None = None
    manufacture_year: int | None = None
    inventory_number: str | None = None
    comment: str | None = None
